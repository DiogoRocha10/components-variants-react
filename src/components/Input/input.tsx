import styles from "./input.module.css"
import cn from "clsx"

interface InputProps {
	label?: string
	value?: string
	variant?: "error" | "success" | "default"
	size?: "sm" | "md" | "lg"
	className?: string
	onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export const Input = ({
	label,
	value,
	variant = "default",
	size = "md",
	className,
	onChange,
	...props
}: InputProps) => {
	const sizeClasses = {
		[styles["-input--small"]]: size === "sm",
		[styles["-input--medium"]]: size === "md",
		[styles["-input--large"]]: size === "lg",
	}

	const variantClasses = {
		[styles["-input--error"]]: variant === "error",
		[styles["-input--success"]]: variant === "success",
		[styles["-input--default"]]: variant === "default",
	}

	return (
		<div className={cn(styles["label-container"])}>
			<label className={cn(className, styles["label"])}>{label}</label>

			<input
				className={cn(className, styles["-input"], variantClasses, sizeClasses)}
				onChange={onChange}
				{...props}
			/>
		</div>
	)
}
