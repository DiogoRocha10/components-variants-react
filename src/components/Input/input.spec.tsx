import { render } from "@testing-library/react"
import { Input } from "./input"
import renderer from "react-test-renderer"

it("text", () => {
	const { getByText } = render(
		<Input label="Default" variant="default" size="md" />
	)
	expect(getByText("Default")).toBeTruthy()
})

it("renders correctly", () => {
	const tree = renderer.create(<Input label="Default" />).toJSON()
	expect(tree).toMatchSnapshot()
})
