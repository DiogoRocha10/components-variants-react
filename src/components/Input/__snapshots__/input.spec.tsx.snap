// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="label-container"
>
  <label
    className="label"
  >
    Default
  </label>
  <input
    className="-input -input--default -input--medium"
  />
</div>
`;
